R/CompDbSource.R

```diff
@@ -162,56 +162,110 @@ MassBankSource <- function(release = "2021.03", ...) {
+#' Experimental (query) fragment spectra can be matched against the MetFrag
+#' server using `matchSpectra` by providing `MatchFragSource()` with parameter
+#' `target`. The parameters for the matching against MetFrag can be specified
+#' using a `MetFragParam` object (parameter `param` of `matchSpectra`).
+#'
+#' Description of functions:
+#'
+#' - `MetFragParam`: creates an `MetFragParam` instance with the parameters
+#'   for the MetFrag query.
+#'
+#' - `MetFragSource`: creates a `MetFragSource` instance and checks if the
+#'   specified URL is valid.
+#'
+#' - `matchSpectra`: annotates provided MS/MS spectra (as a [Spectra()] object
+#'   with parameter `query`) against MetFrag. Polarity and expected adduct
+#'   information have to be set in the provided `Spectra` object: polarity
+#'   has to be encoded with `0` and `1` for negative and positive polarity
+#'   through the spectra variable `"polarity"`. The expected adduct(s) for
+#'   each fragment spectrum need to be defined as a spectra variable
+#'   `"precursorAdduct"`. Use `validAdducts(MetFragParam())` to get the set of
+#'   adduct definitions supported by MetFrag.
+#'   Parameter `target` is expected to be an instance of `MetFragSource`
+#'   providing the URL to the MetFrag server that should be used for the
+#'   matching. Settings for the matching can be provided using an instance of
+#'   `MetFragParam` with parameter `param`. The function iterates over all
+#'   provided query spectra and annotates each using the MetFrag server. Note
+#'   that an active internet connection is required for this function. Note
+#'   also that queries can timeout (in which case a warning is shown and no
+#'   match is reported for the specific fragment spectrum). The result is
+#'   returned as a [Matched()] object with `query` being the input `Spectra`
+#'   and `target` the `data.frame` with matches returned from the MetFrag
+#'   server. See examples below for details.
+#'
+#' - `validAdducts`: `validAdducts(MetFragParam())` returns a `list` with the
+#'   supported adduct names of MetFrag (for positive and negative polarity).
+#'   If called on a [Spectra()] object (`validAdducts(spectra, MetFragParam())`)
+#'   it checks whether the specified adduct names (with spectra variable
+#'   `"precursorAdduct"`) are valid, given the polarity of each spectrum. The
+#'   function returns `TRUE` if all adduct definitions are valid or throws an
+#'   error.
+#'
+#' LLLLL PARAMETERS
+#' @author Nir Shahaf, Johannes Rainer
+NULL
+
+## We Assume that the user has added the precursor adduct annotation(s) to the
+## spectra data, under the column 'precursorAdduct' e.g.:
+## load(system.file("extdata", "sps_test.Rdata",package = "MetaboAnnotation"))
+## sps_test$precursorAdduct <- "[M-H]-"
+## Define MetFrag parameters object and make connection with the online server:
+## mfParam = MetFragParam()
+## mfSrc = MetFragSource()
+##
+## testMatch <- matchSpectra(sps_test, MetFragSource(), MetFragParam())
```

Comment:
just replaced `#'` here with `##` otherwise this is recognized/parsed as roxygen documentation.